### DISCUSSION TITLE
Windows support

### DISCUSSION THREAD

**Discussion by daniel-rikowski** [ElMassimo/vite_ruby#129] [General]
Currently vite_ruby doesn't work on Windows. This is due to `resolveEntrypointFiles` not finding any files, which is caused by:

1. `path.resolve` returning paths in the OS-native format (i.e. on Windows with `\` as path separator)
2. `fast-glob` requiring glob patterns to use only `/` as path separator

Changing this

```ts
const resolvedGlobs = resolveGlobs(projectRoot, sourceCodeDir, inputGlobs)
```
into this 

```ts
const resolvedGlobs = resolveGlobs(projectRoot, sourceCodeDir, inputGlobs).map((pattern) => pattern.replace(/\\/g, '/'))
```

makes vite_ruby work on Windows. (at least it looks that way 🙏 )

I can make a pull request, but since the change is trivial - and perhaps too trivial - I'd like to discuss it first 🤞

**Comment by ElMassimo** [comment]
Hi Daniel! Thanks for sharing!

Would you be willing to test a pre-release of `vite-plugin-ruby` with the suggested fix?

@daniel-rikowski Released as `vite-plugin-ruby@3.0.2-pre.1`

**Comment by daniel-rikowski** [comment]
I tried but it doesn't work. Now `resolveGlobs` returns this:

```js
[
 "C:\Projects\MyProject/C:/Projects/MyProject/C:\Projects\MyProject\app\frontend/entrypoints/**/*",
 "C:\Projects\MyProject/C:/Projects/MyProject/C:\Projects\MyProject\app\frontend/{assets,fonts,icons,images}/**/*",
]
```

with
```js
projectRoot = "C:\Projects\MyProject"
sourceCodeDir = "C:\Projects\MyProject\app\frontend"
```

It looks like both parameters to `path.posix.resolve` get prepended with the current working directory.


This might be related to a long-standing bug in node regarding `path.posix`. (https://github.com/nodejs/node/issues/13887) 
I'm using node 14.18.0 where it should be fixed, but perhaps this is caused by something else.

Both of these lines work, though:

```js
slash(path.win32.resolve(projectRoot, pattern.replace(/^~\//, `${sourceCodeDir}/`)))
slash(path.resolve(projectRoot, pattern.replace(/^~\//, `${sourceCodeDir}/`)))
```

**Reply by ElMassimo** [reply]
Thanks for testing it, used the second line of your example, it's now released as `vite-plugin-ruby@3.0.2-pre.2`.

`sourceCodeDir` should be relative to the root and use POSIX separators: `app/frontend`.

**Comment by daniel-rikowski** [comment]
It works now! Thank you for the quick response!

**Reply by ElMassimo** [reply]
I'll make a `3.0.2` release during the week.

Thanks for reporting and helping with the fix! 😃
